File: recursion/code_samples/php/fibonacci.php
<?php

function fibonacciNumber($n) {
    if ($n == 0 || $n == 1) {
        return $n;
    }
    return (fibonacciNumber($n-2) + fibonacciNumber($n-1));
}

$x = fibonacci(10);
echo "$x\n";
?>

File: recursion/code_samples/java/fibonacci.java
package java;

public class fibonacci {

    static int fibonacciNumber(int n) {
        if (n == 0 || n == 1) {
            return n;
          } else {
            return fibonacciNumber(n-2) + fibonacciNumber(n-1);
          }
    }
    
    public static void main(String[] args) {
        int x = fibonacciNumber(10);
        System.out.println(x);
    }
}
